e2e/cypress/integration/enterprise/teams/search_teams_spec.js

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import uuid from 'uuid/v4';
+const PAGE_SIZE = 10;
+
+describe('Search teams', () => {
+    beforeEach(() => {
+        cy.apiLogin('sysadmin');
+        cy.visit('/admin_console/user_management/teams');
+    });
+
+    it('loads with no search text', () => {
+        // * Check the search input is empty.
+        cy.findByTestId('search-input').should('be.visible').and('have.text', '');
+    });
+
+    it('returns results', () => {
+        const displayName = uuid();
+
+        // # Create a new team.
+        cy.apiCreateTeam('team-search', displayName);
+
+        // # Search for the new team.
+        cy.findByTestId('search-input').type(displayName + '{enter}');
+
+        // * Check that the search results contain the team.
+        cy.findAllByTestId('team-display-name').contains(displayName);
+    });
+
+    it('results are paginated', () => {
+        const displayName = uuid();
+
+        // # Create enough new teams with common name prefixes to get multiple pages of search results.
+        for (let i = 0; i < PAGE_SIZE + 2; i++) {
+            cy.apiCreateTeam('team-search-paged-' + i, displayName + ' ' + i);
+        }
+
+        // # Search using the commot team name prefix.
+        cy.findByTestId('search-input').type(displayName + '{enter}');
+
+        // * Check that the first page of results is full.
+        cy.findAllByTestId('team-display-name').should('have.length', PAGE_SIZE);
+
+        // # Click the next pagination arrow.
+        cy.findByTestId('page-link-next').should('be.enabled').click();
+
+        // * Check that the 2nd page of results has the expected amount.
+        cy.findAllByTestId('team-display-name').should('have.length', 2);
+    });
+
+    it('clears the results when "x" is clicked', () => {
+        const displayName = uuid();
+
+        // # Create a new team.
+        cy.apiCreateTeam('team-search', displayName);
+
+        // # Search for the team.
+        cy.get('[data-testid=search-input]').as('searchInput').type(displayName + '{enter}');
```

Comment:
@srkgupta Note that the `findByTestId` function breaks the ability to set an alias.

Reply:
I'm going to look into it and possibly file a bug with `cypress-testing-library`.